src/app.rs

```diff
@@ -191,35 +201,38 @@ impl<'repo> App<'repo> {
         f.render_widget(content, area);
     }
 
-    pub fn navigate(&mut self, action: NavigationAction) -> Result<(), GitBrowserError> {
+    pub fn navigate(&mut self, action: &NavigationAction) -> Result<bool, GitBrowserError> {
```

Comment:
What does this `bool` represent, does it mean "navigation failed/succeeded"?

Would it be more readable with an enum? `enum NavigationStatus { Success; Failure}`, or `struct NavigationSuccess(bool)`?

Returning a primitive type might not convey the purpose well enough, what do you think?

Reply:
I don't see this as a problem/a blocking issue, just thinking if the return type could have more contextual information to describe its purpose better.

Reply:
It's to force a full screen redraw. After returning to "normal" terminal output and launching the external pager, we have to do a full screen clear and full redraw, otherwise the UI is corrupted.

I agree some kind of enum is better than an obscure bool, but I'm not quite sure what that should be called then.

Reply:
"struct RedrawRequired(bool)" or "struct ForceRedraw(bool)" could be an option